### PR TITLE
Moving LSTM from alt-modular to Public NOAA 

### PR DESCRIPTION
**Pull request by jmframe**
[Short description explaining the high-level reason for the pull request]

## Additions

- everything

## Removals

- The out-of-date model that was not easily adaptable to new trained weights.

## Changes

-

## Testing

1.

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,105 +1,47 @@
...
-
-----
-
-## Credits and references
-
-1. Projects that inspired you
-2. Related projects
-3. Books, papers, talks, or other sources that have meaningful impact or influence on this project
+# Basic Model Interface (BMI) for streamflow prediction using Long Short-Term Memory (LSTM) networks
+This LSTM is for use in the Next Generation National Water Model. LSTMs have been shown to be very good deep learning models for streamflow prediction. This module is available through a BMI interface that is build directly into the deep learning model class. That means this LSTM is inherently BMI enabled.
```
Comment (madMatchstick) [inline README.md:2]: built

File: README.md [hunk 2]
```diff
@@ -1,105 +1,47 @@
...
+
+# Model configuration
+The LSTM model requires a configuration file for specification of forcings, weights, scalers, run options (like warmup period), run time period, static basin parameters and model time step. This configuration file needs to be generated for any specific application of the LSTM model.
+This LSTM model will run on any basin with the required inputs. The place to set up the run for a specific configuration for a specific basin is in the BMI configuration file: `./bmi_config_files/*.yml`.
+
+# Trained LSTM model
+Included in this directory are two sample of trained LSTM models:
+* hourly_A
+* hourly_B  
+These two models are trained with slightly different inputs (Basin average slope is included in only one), and with different number of hiddel layers.
```
Comment (madMatchstick) [inline README.md:26]: hidden?

File: README.md [hunk 3]
```diff
@@ -1,105 +1,47 @@
...
+
+# System requirements to run this model
+Running this model requires python and the libraries listed in the environment file: `environment.yml`.  
+If you can load in the environments without Anaconda, that should be just fine. Notice that `xarray` has a specific version defined in the environment file (0.14.0). The newer versions are uncompatible withe the example files, for some reason that will be fixed at some point, so for now, make sure to use this specific version of Xarray.
```
Comment (madMatchstick) [inline README.md:30]: incompatible with

File: README.md [hunk 4]
```diff
@@ -1,105 +1,47 @@
...
+# Trained LSTM model
+Included in this directory are two sample of trained LSTM models:
+* hourly_A
+* hourly_B  
+These two models are trained with slightly different inputs (Basin average slope is included in only one), and with different number of hiddel layers.
+
+# System requirements to run this model
+Running this model requires python and the libraries listed in the environment file: `environment.yml`.  
+If you can load in the environments without Anaconda, that should be just fine. Notice that `xarray` has a specific version defined in the environment file (0.14.0). The newer versions are uncompatible withe the example files, for some reason that will be fixed at some point, so for now, make sure to use this specific version of Xarray.  
+It may be easiest to make sure a python environment containing pytorch, and the others listed above, are installed and activated with Anaconda using: `conda env create -f environment.yml`. This give you a conda environment called `bmi_lstm`.
```
Comment (madMatchstick) [inline README.md:31]: gives

File: README.md [hunk 5]
```diff
@@ -1,105 +1,47 @@
...
+3. Read in the configuration file, and this includes the model weights, etc.: `model.read_cfg_file()`
+4. Now start running the BMI functions, starting with initialize: `model.initialize()`
+5. The model is now available to run either one timestep at a time: `model.update()`, or many timesteps at a time: `model.update_until(model.iend)`, where model.iend is the end of the forcing file, but this can be any value less than or equal to the end of the forcing file.
+6. And finally you should finalize the model instance: `model.finalize()`
+
+# Model weights and biases
+The training procedure should produce weights and biases for the LSTM model. These are stored in Pytorch files (*.pt), are kept within the training directories: `trained_neuralhydrology_models`. Without these the model can still run, but will not make streamflow predictions. These are **absolutely** neccessary for running this model with NextGen. These weights and biases are trained to represent many basins, so they do not change for every basins. The model may be trained regionally, or globally, and the weights and biases need to be consistent across the appropriate basins.
```
Comment (madMatchstick) [inline README.md:44]: necessary

Comment (madMatchstick) [inline README.md:44]: so they do not change for every basin.

File: README.md (hunk 3 above)
Comment (madMatchstick) [inline README.md:30]: Suggestion: The newer versions are incompatible with how `pickle` stores data loaded from example files.

File: README.md [hunk 6]
```diff
@@ -1,105 +1,47 @@
...
+* 03015500 Brokenstraw Creek at Youngsville, PA
+* 01022500 Narraguagus River at Cherryfield, Maine  
+These are just samples, you can run on any watershed.   
+
+# Model configuration
+The LSTM model requires a configuration file for specification of forcings, weights, scalers, run options (like warmup period), run time period, static basin parameters and model time step. This configuration file needs to be generated for any specific application of the LSTM model.
+This LSTM model will run on any basin with the required inputs. The place to set up the run for a specific configuration for a specific basin is in the BMI configuration file: `./bmi_config_files/*.yml`.
+
+# Trained LSTM model
+Included in this directory are two sample of trained LSTM models:
```
Comment (madMatchstick) [inline README.md:23]: samples

**Review by madMatchstick — APPROVED** [review]
Loaded in `environment.yml` via Anaconda on windows 10 with no errors.  Both `hourly_A` and `hourly_B` sample sets run fine. 

See minor suggestions (mostly typos) in README.  Otherwise, looks good!

File: README.md (hunk 5 above)
Comment (jmframe) [reply README.md:44]: Correct.

File: README.md (hunk 3 above)
Comment (jmframe) [reply README.md:30]: It is actually an Xarray version issue. Pickle serializes the file in whatever way that Xarray version does it, so if you use a new Xarray version, then Pickle will just unpack the old version, and the conflict is just with Xarray.

**Pull request merged**